### PR TITLE
Update repo-config from v0.9.2 to v0.10.0

### PR DESCRIPTION
**Pull request by ghost**
Update repo-config following the instructions described [here](https://frequenz-floss.github.io/frequenz-repo-config-python/latest/user-guide/update-an-existing-project/).

### CODE REVIEW

File: .github/containers/test-installation/Dockerfile [hunk 1]
```diff
@@ -13,5 +13,15 @@ RUN apt-get update -y && \
...
+# This git-credentials file is made available by the GitHub ci.yaml workflow
+COPY git-credentials /root/.git-credentials
+RUN git config --global credential.helper store && \
+pip install dist/*.whl && \
+rm -rf dist /root/.git-credentials
+# This git-credentials file is made available by the GitHub ci.yaml workflow
+COPY git-credentials /root/.git-credentials
+RUN git config --global credential.helper store && \
+pip install dist/*.whl && \
+rm -rf dist /root/.git-credentials
```
Comment (llucax) [inline .github/containers/test-installation/Dockerfile]: This got duplicated for some reason.

```suggestion
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -41,13 +41,55 @@ jobs:
...
+        # with:
+        #   username: ${{ secrets.GIT_USER }}
+        #   password: ${{ secrets.GIT_PASS }}
+
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+        # TODO(cookiecutter): Uncomment this for projects with private dependencies
+        # with:
+        #   username: ${{ secrets.GIT_USER }}
+        #   password: ${{ secrets.GIT_PASS }}
```
Comment (llucax) [inline .github/workflows/ci.yaml]: **Everything** got duplicated for some reason! :rofl:  (will not comment on duplicated code anymore)

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -41,13 +41,55 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+        # TODO(cookiecutter): Uncomment this for projects with private dependencies
+        # with:
+        #   username: ${{ secrets.GIT_USER }}
+        #   password: ${{ secrets.GIT_PASS }}
```
Comment (llucax) [inline .github/workflows/ci.yaml]: You can remove these comments.

```suggestion
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -87,355 +129,12 @@ jobs:
   nox-all:
     # The job name should match the name of the `nox` job.
     name: Test with nox
-    needs: ["nox"]
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Return true
-        run: "true"
-
-  nox-cross-arch:
```
Comment (llucax) [inline .github/workflows/ci.yaml]: All of these (including the whole `nox-cross-arch` job) should not be gone.

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -87,355 +129,12 @@ jobs:
...
-  # https://github.com/orgs/community/discussions/9141
-  nox-cross-arch-all:
-    # The job name should match the name of the `nox-cross-arch` job.
-    name: Cross-arch tests with nox
     needs: ["nox-cross-arch"]
+    # We skip this job only if nox-cross-arch was also skipped
+    if: always() && needs.nox-cross-arch.result != 'skipped'
     runs-on: ubuntu-20.04
+    env:
+      DEPS_RESULT: ${{ needs.nox-cross-arch.result }}
```
Comment (llucax) [inline .github/workflows/ci.yaml:219]: These updates are the right ones.

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -87,355 +129,12 @@ jobs:
...
     steps:
-      - name: Return true
-        run: "true"
-
-  build:
```
Comment (llucax) [inline .github/workflows/ci.yaml:210]: Also all the following jobs should not be gone.

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -87,355 +129,12 @@ jobs:
...
-      - name: Download distribution files
-        uses: actions/download-artifact@v3
-        with:
-          name: dist-packages
-          path: dist
-
-      - name: Publish the Python distribution to PyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
+      - name: Check matrix job result
+        run: test "$DEPS_RESULT" = "success"
```
Comment (llucax) [inline .github/workflows/ci.yaml:222]: This is also part of the actual update (update to the `nox-cross-arch-all` job.

File: mkdocs.yml [hunk 8]
```diff
@@ -110,7 +110,7 @@ plugins:
             show_root_members_full_path: true
             show_signature_annotations: true
             show_source: true
-            signature_crossrefs: true
+            show_symbol_type_toc: true            show_symbol_type_toc: true            show_symbol_type_toc: true            show_symbol_type_toc: true            show_symbol_type_toc: true            show_symbol_type_toc: true            show_symbol_type_toc: true            show_symbol_type_toc: true            signature_crossrefs: true
```
Comment (llucax) [inline mkdocs.yml]: This got fucked up.
```suggestion
            show_symbol_type_toc: true
            signature_crossrefs: true
```

File: pyproject.toml [hunk 9]
```diff
@@ -59,7 +59,7 @@ dev-mkdocs = [
   "mkdocs-macros-plugin == 1.0.5",
   "mkdocs-material == 9.5.25",
   "mkdocstrings[python] == 0.25.1",
-  "frequenz-repo-config[lib] == 0.9.2",
+  "mkdocstrings-python == 1.9.2",  "mkdocstrings-python == 1.9.2",  "mkdocstrings-python == 1.9.2",  "mkdocstrings-python == 1.9.2",  "mkdocstrings-python == 1.9.2",  "mkdocstrings-python == 1.9.2",  "mkdocstrings-python == 1.9.2",  "frequenz-repo-config[lib] == 0.9.2",
```
Comment (llucax) [inline pyproject.toml]: Also fucked up, it should be only one.

File: pyproject.toml [hunk 10]
```diff
@@ -140,8 +140,36 @@ disable = [
...
+  "f-string-without-interpolation",
+  "f-string-without-interpolation",
   "line-too-long",
+  "missing-function-docstring",
+  "missing-function-docstring",
+  "missing-function-docstring",
+  "missing-function-docstring",
+  "missing-function-docstring",
+  "missing-function-docstring",
+  "missing-function-docstring",
```
Comment (llucax) [inline pyproject.toml]: Also many dupes.

**Review by llucax — COMMENTED** [review]
I guess you applied the migration script many times due to the macOS problems, so you got a lot of duplication.

**Comment by ghost** [comment]
Yep what happened is that I was fixing the script as I went and iteratively re-running it for each bug. I indeed didn't notice that it was re-writting stuff each time! This should be fixed now. I however still get an error from the Dockerfile

File: .github/containers/test-installation/Dockerfile [hunk 11]
```diff
@@ -13,5 +13,10 @@ RUN apt-get update -y && \
     python -m pip install --upgrade --no-cache-dir pip
 
 COPY dist dist
+# This git-credentials file is made available by the GitHub ci.yaml workflow
+COPY git-credentials /root/.git-credentials
+RUN git config --global credential.helper store && \
+pip install dist/*.whl && \
+rm -rf dist /root/.git-credentials
```
Comment (llucax) [inline .github/containers/test-installation/Dockerfile]: Shouldn't fix any erros, but looks more readable like this:

```suggestion
    pip install dist/*.whl && \
    rm -rf dist /root/.git-credentials
```

File: .github/workflows/ci.yaml [hunk 12]
```diff
@@ -41,13 +41,22 @@ jobs:
...
     steps:
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+
       - name: Print environment (debug)
         run: env
 
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+
```
Comment (llucax) [inline .github/workflows/ci.yaml]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 13]
```diff
@@ -41,13 +41,22 @@ jobs:
...
       - name: Fetch sources
         uses: actions/checkout@v4
         with:
           submodules: true
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+
```
Comment (llucax) [inline .github/workflows/ci.yaml]:
```suggestion
```

File: .github/workflows/ci.yaml
Comment (llucax) [inline .github/workflows/ci.yaml:1]: Still some dupes here.

File: .github/workflows/ci.yaml [hunk 14]
```diff
@@ -115,10 +128,16 @@ jobs:
...
```
Comment (llucax) [inline .github/workflows/ci.yaml]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 15]
```diff
@@ -202,19 +221,30 @@ jobs:
...
```
Comment (llucax) [inline .github/workflows/ci.yaml]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 16]
```diff
@@ -268,10 +307,16 @@ jobs:
...
```
Comment (llucax) [inline .github/workflows/ci.yaml]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 17]
```diff
@@ -310,10 +355,15 @@ jobs:
...
     steps:
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+
       - name: Fetch sources
         uses: actions/checkout@v4
         with:
           submodules: true
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
```
Comment (llucax) [inline .github/workflows/ci.yaml]:
```suggestion
```

File: pyproject.toml [hunk 18]
```diff
@@ -59,7 +59,7 @@ dev-mkdocs = [
   "mkdocs-macros-plugin == 1.0.5",
   "mkdocs-material == 9.5.25",
   "mkdocstrings[python] == 0.25.1",
-  "frequenz-repo-config[lib] == 0.9.2",
+  "mkdocstrings-python == 1.9.2",  "frequenz-repo-config[lib] == 0.9.2",
```
Comment (llucax) [inline pyproject.toml]:
```suggestion
  "mkdocstrings-python == 1.9.2",
  "frequenz-repo-config[lib] == 0.10.0",
```

And you should search for the other instances of `frequenz-repo-config` and update it to `0.10.0` (there are 3 or 4 in total in this file).

File: pyproject.toml [hunk 19]
```diff
@@ -123,7 +124,7 @@ check-yield-types = false
 arg-type-hints-in-docstring = false
 arg-type-hints-in-signature = true
 allow-init-docstring = true
-
+check-class-attributes = false
```
Comment (llucax) [inline pyproject.toml:127]: Nitpick 

```suggestion
check-class-attributes = false

```

**Review by llucax — DISMISSED** [review]

**Review by llucax — APPROVED** [review]

**Pull request merged**